### ISSUE TITLE
Loss function

### ISSUE THREAD

**Issue by yufansong** [WuJie1010/Temporally-language-grounding#4]
1. In the code, you didn't add Location/IoU loss, what you do is just create the variables without assigning any value. Actually, if I try to revise the loss function into correct code. I find the performance will decrease largely.

2. The policy loss function is always negative and what we want to do is minimum this loss rather than make it always increase. I was wondering the reason is reward value is always lower than state value(According to the A2C formula) or this is the bug of the code.

**Comment by WuJie1010** [comment]
1.  (policy_loss + value_loss + iou_loss + loc_loss).backward(retain_graph = True) have added the loss. I find it achieves a comparable result with/without these two losses.
2. It is an interesting phenomenon， I guess the reason is the reward value is always lower than state value.

**Comment by yufansong** [comment]
I mean you only have 
```
locations = torch.zeros(opt.num_steps, batch_size, 2)
```
but without 
```
locations[step, :] = location
```
so this part will have bug
```
loc_loss += (torch.abs(offset_norm[j][0].cpu() - locations[i][j][0]) + torch.abs(offset_norm[j][1].cpu() - locations[i][j][1]))
```

**Comment by yufansong** [comment]
For the policy loss part, I agree with you. I think the accumulated reward is too small, so the R is always lower than the state value.

**Comment by WuJie1010** [comment]
> I mean you only have
> 
> ```
> locations = torch.zeros(opt.num_steps, batch_size, 2)
> ```
> 
> but without
> 
> ```
> locations[step, :] = location
> ```
> 
> so this part will have bug
> 
> ```
> loc_loss += (torch.abs(offset_norm[j][0].cpu() - locations[i][j][0]) + torch.abs(offset_norm[j][1].cpu() - locations[i][j][1]))
> ```

oh~ your are right~ I will fix this bug~

**Comment by yufansong** [comment]
Also, the predict IoU should be fixed.
```
Predict_IoUs[step, :] = tIoU
```

**Comment by yufansong** [comment]
Your reply is so quick. I just was trying to pull the request to fix those two bugs, but you have already fixed one of it.

**Comment by WuJie1010** [comment]
> Your reply is so quick. I just was trying to pull the request to fix those two bugs, but you have already fixed one of it.

Thanks for your suggestions~ you can pull  the request if you find other bugs~ Thanks~
